### DISCUSSION TITLE
How to use the tag feature of Postal

### DISCUSSION THREAD

**Discussion by jazer143** [postalserver/postal#2006] [Help with using Postal] [answered]
Hi There,

Newbie here, seeking kind help who's  familiar on how to use/insert a tag in an email for filtering purposes. 

Thank you.

**Comment by willpower232** [comment, marked as answer]
Postal doesn't hold onto the messages for very long so if you want to manage emails, you should do that in whatever system is sending the emails.

If you are able to create the message for SMTP then you should be able to add whatever you want to the message.

**Reply by jazer143** [reply]
> Postal doesn't hold onto the messages for very long so if you want to manage emails, you should do that in whatever system is sending the emails.
> 
> If you are able to create the message for SMTP then you should be able to add whatever you want to the message.

Hi @willpower232 , Thank you for taking time to reply on my query. I have the resources to make the email going in/out stays longer in the server. I relayed emails from one application to Postal, so the tag should be added in the source right? I thought Postal could somehow alter or insert a header tag. I don't think my application supports header modification.

My issue is that, I can't filter the email based on the header from or by domain. The Postal filter feature is limited and currently filter by _envelop from_. Our application is using one envelop from and then we use different header from and I enabled Postal setting's for Allow sender header . Hope that in the future will add more filtering like by domain, etc. Thank you.

**Reply by willpower232** [reply]
Even if you configure Postal to retain emails for a long period of time, it isn't really designed to act this way.

Besides, all the context for sending the message would be in the application which created the message in the first place so you would definitely be better served by dealing with this in whichever application you're using to send the messages.
